CardPayments/src/main/java/com/paypal/android/cardpayments/CardClient.kt

```diff
@@ -129,24 +117,23 @@ class CardClient internal constructor(
         if (browserSwitchResult != null && approveOrderListener != null) {
             approveOrderListener?.onApproveOrderThreeDSecureDidFinish()
             when (browserSwitchResult.status) {
-                BrowserSwitchStatus.SUCCESS -> getOrderInfo(browserSwitchResult)
+                BrowserSwitchStatus.SUCCESS -> handleBrowserSwitchSuccess(browserSwitchResult)
                 BrowserSwitchStatus.CANCELED -> notifyApproveOrderCanceled()
             }
         }
     }
 
-    private fun getOrderInfo(browserSwitchResult: BrowserSwitchResult) {
+    private fun handleBrowserSwitchSuccess(browserSwitchResult: BrowserSwitchResult) {
         ApproveOrderMetadata.fromJSON(browserSwitchResult.requestMetadata)?.let { metadata ->
             CoroutineScope(dispatcher).launch(exceptionHandler) {
                 try {
-                    val getOrderResponse = checkoutOrdersAPI.getOrderInfo(GetOrderRequest(metadata.orderID))
                     analyticsService.sendAnalyticsEvent(
                         "card-payments:3ds:get-order-info:succeeded",
                         metadata.orderID
                     )
                     val deepLinkUrl = browserSwitchResult.deepLinkUrl
-                    val result = getOrderResponse.run {
-                        CardResult(orderId, orderStatus, paymentSource, deepLinkUrl)
+                    val result = metadata.run {
+                        CardResult(orderID, paymentSource, deepLinkUrl)
```

Comment:
isnt PaymentSource here null? After doing `confirm_payment_source`, we dont get the actual payment source, we were getting it from the `getorderinfo`. So i think that we can only return `orderID` and `deeplinkurl`

Reply:
Yeah good call, this is `null` in the 3DS flow but not when 3DS is skipped it has a value.

I'm not sure what to do in that case. Client ID's will limit functionality going forward. We may need to reach out to the team that owns the GetOrder API to allow client-id-only access.

Reply:
After discussion, it sounds like the right move may be to remove the `paymentSource` property from `CardResult`. Merchants can get this information from authorize and capture server-side calls.

Reply:
Agreed - [for iOS](https://github.com/paypal/iOS-SDK/blob/be9ae349d5c1dc3cdd4eab56f57f86003b126589/Sources/CardPayments/Models/CardResult.swift#L12-L16) we'll have to remove both `status` and `paymentSource`.

We should also align these types across platforms - does the merchant really need the [deepLinkURI on Android](https://github.com/paypal/Android-SDK/blob/a3b97cd5ed7d1f7c2639d98e6fb1ea44309639b2/CardPayments/src/main/java/com/paypal/android/cardpayments/model/CardResult.kt#LL13C1-L13C1)?

Reply:
Also, there might be valuable info in the redirectURL we could parse and include on the result. I know @KunJeongPark started that investigation.

Reply:
Right yeah, if I recall we didn't parse the deep link URL when we first implemented this feature because the callback parameters weren't very uniform.

Reply:
From 2 test cards I tried, I didn't see anything informative from the url returned from 3DS session. I can try more cards and ask 3DS folks.

Reply:
Dang, that's disappointing that the sandbox URLs don't have much useful info and are inconsistent between test cards. Agreed, we should reach out to whoever owns this to learn more.

I found a convo in Slack from last year where Don Roman recommends using the test card generator for potentially more accurate test cards:
> In sandbox, one can generate test cards using credit card generator. Please login to developer dashboard: https://developer.paypal.com/developer/applications and find the credit card generator on the left panel.

Reply:
The test card generator looks like it's in JavaScript I wonder if we could do something similar. All the test cards start with `403203` and the rest might be a random number that makes it Luhn valid.